js/src/provider/WalletLinkProvider.ts

```diff
@@ -650,45 +651,45 @@ export class WalletLinkProvider implements Web3Provider {
     params: unknown[]
   ): Promise<JSONRPCResponse> {
     this._requireAuthorization()
-    const typedData = params[0]
+    const typedDataJson = ensureJsonOrParseString(params[0])
     const address = ensureAddressString(params[1])
 
     this._ensureKnownAddress(address)
 
-    const message = eip712.hashForSignTypedDataLegacy({ data: typedData })
-    const typedDataJson = JSON.stringify(typedData, null, 2)
+    const message = eip712.hashForSignTypedDataLegacy({ data: typedDataJson })
+    const typedDataStr = JSON.stringify(typedDataJson, null, 2)
```

Comment:
and this can be called `typedDataJson`

Reply:
c5e0d171e4a6fa75f11ce18f934709047f095034